runtime/util/vmargs.c

```diff
@@ -76,8 +76,41 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+#define JAVA_CLASS_PATH_EQUALS "-Djava.class.path="
+#define SUN_JAVA_COMMAND_EQUALS "-Dsun.java.command="
+#define SUN_JAVA_LAUNCHER_PID_EQUALS "-Dsun.java.launcher.pid="
+#define SUN_JAVA_LAUNCHER_EQUALS "-Dsun.java.launcher="
+
+struct JavaLauncherOption {
+	const char *optionName;
+	int optionNameLength;
+};
+
+/* Options in this table will be not treated as user arguments */
+static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
+	{JAVA_CLASS_PATH_EQUALS, LITERAL_STRLEN(JAVA_CLASS_PATH_EQUALS)},
+	{SUN_JAVA_COMMAND_EQUALS, LITERAL_STRLEN(SUN_JAVA_COMMAND_EQUALS)},
+	{SUN_JAVA_LAUNCHER_PID_EQUALS, LITERAL_STRLEN(SUN_JAVA_LAUNCHER_PID_EQUALS)},
+	{SUN_JAVA_LAUNCHER_EQUALS, LITERAL_STRLEN(SUN_JAVA_LAUNCHER_EQUALS)},
+	{VMOPT_XJCL_COLON, LITERAL_STRLEN(VMOPT_XJCL_COLON)}, /* macro from jvminit.h */
+	{NULL, 0} /* The last entry in the table must be Null in order for the search function to exit.*/
+};
+
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
+/* Binary Search function that searches if optionName is in JavaLauncherOptionTable */
+BOOLEAN
+isGeneratedArgument(const char *optionName) {
+	const struct JavaLauncherOption *tableOptionName = JavaLauncherOptionTable;
+	while (tableOptionName->optionName != NULL) {
+		if (0 == strncmp(optionName, tableOptionName->optionName, tableOptionName->optionNameLength)) {
```

Comment:
strncmp will match substrings, use `J9UTF8_DATA_EQUALS` instead

Reply:
We need it to check the substring since some arguments might have parameters (EG: -Djava.class.path=/User/...). If we did an exact comparison, we wouldn't get a match unless the user argument is the exact same as the one in the table